pkg/cluster/cert.go

```diff
@@ -32,9 +32,25 @@ func CreateSelfSignedCertificate(ctx context.Context, c client.Client, secretNam
 	if err := ApplyMetaOptions(certSecret, metaOptions...); err != nil {
 		return err
 	}
-
-	if createErr := c.Create(ctx, certSecret); client.IgnoreAlreadyExists(createErr) != nil {
-		return fmt.Errorf("failed creating certificate secret: %w", createErr)
+	existingSecret := &corev1.Secret{}
+	err = c.Get(ctx, client.ObjectKey{Name: secretName, Namespace: namespace}, existingSecret)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// Secret does not exist, create it
+			if createErr := c.Create(ctx, certSecret); createErr != nil {
+				return fmt.Errorf("failed creating certificate secret: %w", createErr)
+			}
+		} else {
+			return fmt.Errorf("failed getting certificate secret: %w", err)
+		}
+	} else if existingSecret.Type != certSecret.Type {
```

Comment:
Isn't it a candidate for recreateSecret() as well?

Reply:
:+1: you are right, this block should use recreateSecret() call. i will make a follow up

Reply:
https://github.com/opendatahub-io/opendatahub-operator/pull/1080

Reply:
Just want to mention, that the merging of the PR was premature, we haven't finish discussion.

Reply:
Yep it got auto merged. Lets move the discussion here https://github.com/opendatahub-io/opendatahub-operator/pull/1080